test/integration/weighted_cluster_integration_test.cc

```diff
@@ -0,0 +1,209 @@
+#include <cstdint>
+#include <iterator>
+#include <numeric>
+
+#include "envoy/config/bootstrap/v3/bootstrap.pb.h"
+#include "envoy/config/route/v3/route_components.pb.h"
+#include "envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.pb.h"
+
+#include "test/integration/filters/repick_cluster_filter.h"
+#include "test/integration/http_integration.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+// For the simplicity of testing purpose, the pattern of the cluster array is below:
+// [0, ClusterNameClusterCount) will be clusters with `name` field specified.
+// [ClusterNameClusterCount, ClusterCount) will be clusters with `cluster_header` field specified.
+constexpr int ClusterCount = Envoy::RepickClusterFilter::ClusterCount;
+constexpr int ClusterNameClusterCount = Envoy::RepickClusterFilter::ClusterNameClusterCount;
+
+class WeightedClusterIntegrationTest : public testing::Test, public HttpIntegrationTest {
+public:
+  WeightedClusterIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP2, Network::Address::IpVersion::v6) {
+    default_weights_.reserve(ClusterCount);
+    std::fill_n(std::back_inserter(default_weights_), ClusterNameClusterCount, 20);
+    std::fill_n(std::back_inserter(default_weights_),
+                Envoy::RepickClusterFilter::ClusterHeaderClusterCount, 30);
+  }
+
+  void createUpstreams() override {
+    setUpstreamProtocol(FakeHttpConnection::Type::HTTP2);
+    //  Add fake upstreams
+    for (int i = 0; i < ClusterCount; ++i) {
+      addFakeUpstream(FakeHttpConnection::Type::HTTP2);
+    }
+  }
+
+  void initializeConfig(const std::vector<uint64_t>& weights) {
+    config_helper_.addConfigModifier([this](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+      // It starts from 1 here because the first cluster is existing cluster configured by
+      // `mutable_clusters` below.
+      for (int i = 1; i < ClusterCount; ++i) {
+        auto* cluster = bootstrap.mutable_static_resources()->add_clusters();
+        cluster->MergeFrom(bootstrap.static_resources().clusters()[0]);
+        cluster->mutable_load_assignment()
```

Comment:
You do not need to set the address or protocol options. Framework will take care of this. All you need is this:
```
bootstrap.mutable_static_resources()->mutable_clusters()->Add()->MergeFrom(
    *bootstrap.mutable_static_resources()->mutable_clusters(0));
bootstrap.mutable_static_resources()->mutable_clusters(1)->set_name("cluster_1");
```